src/NovelBasicFillRect.cpp

```diff
@@ -77,14 +82,8 @@ void NovelBasicFillRect::configureObjectBuffers(bool refreshBuffers) {
       rScalar, gScalar, bScalar, aScalar,
   };
 
-  if(refreshBuffers) {
-    glGenBuffers(1, &_colourBuffer);
-    glBindBuffer(GL_ARRAY_BUFFER, _colourBuffer);
-  }
-
+  auto colourBuffer = _colourBuffer.getActual();
+  glBindBuffer(GL_ARRAY_BUFFER, colourBuffer);
   glBufferData(GL_ARRAY_BUFFER, sizeof(GLfloat) * _colourData.size(), _colourData.data(), GL_STATIC_DRAW);
 }
-NovelBasicFillRect::~NovelBasicFillRect() {
```

Comment:
If you expose a `Lazy<T>.isCreated()` method, you can do:
```
if (_vertexArrayObject.isCreated())
{
    glDeleteVertexArrays(1, &_vertexArrayObject.getActual());
}
```

Reply:
Would probably be good to start adding ogl cleanup everywhere now

Reply:
Unfortunately I can't use a reference on an rvalue of type `GLuint` but ill add the `isCreated()` check.

Reply:
```
auto temp = _vertexArrayObject.getActual();
glDeleteVertexArrays(1, &temp);
_vertexArrayObject.reset();
```

Reply:
is the reset call actually required? Also this functionality was moved to the base render object type.

Reply:
no, probably not. In the case of C++, the finalizer is called under object deletion scenario, so the object won't stay around (unlike C#).